AdobeStockAsset/Model/SaveAsset.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeStockAsset\Model;
+
+use Magento\AdobeStockAssetApi\Api\AssetRepositoryInterface;
+use Magento\AdobeStockAssetApi\Api\CategoryRepositoryInterface;
+use Magento\AdobeStockAssetApi\Api\CreatorRepositoryInterface;
+use Magento\AdobeStockAssetApi\Api\Data\AssetInterface;
+use Magento\Framework\Exception\AlreadyExistsException;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Class SaveAsset
+ *
+ * Service for saving asset
+ */
+class SaveAsset
+{
+    /**
+     * @var AssetRepositoryInterface
+     */
+    private $assetRepository;
+
+    /**
+     * @var CategoryRepositoryInterface
+     */
+    private $categoryRepository;
+
+    /**
+     * @var CreatorRepositoryInterface
+     */
+    private $creatorRepository;
+
+    /**
+     * SaveAsset constructor.
+     *
+     * @param AssetRepositoryInterface $assetRepository
+     * @param CreatorRepositoryInterface $creatorRepository
+     * @param CategoryRepositoryInterface $categoryRepository
+     */
+    public function __construct(
+        AssetRepositoryInterface $assetRepository,
+        CreatorRepositoryInterface $creatorRepository,
+        CategoryRepositoryInterface $categoryRepository
+    ) {
+        $this->assetRepository = $assetRepository;
+        $this->creatorRepository = $creatorRepository;
+        $this->categoryRepository = $categoryRepository;
+    }
+
+    /**
+     * Save asset
+     *
+     * @param AssetInterface $asset
+     * @throws AlreadyExistsException
+     */
+    public function execute(AssetInterface $asset): void
+    {
+        $category = $asset->getCategory();
+        if ($category !== null && !$this->isCategorySaved($category->getId())) {
+            $category = $this->categoryRepository->save($category);
+        }
+        $asset->setCategoryId($category->getId());
+
+        $creator = $asset->getCreator();
+        if ($creator !== null && !$this->isCreatorSaved($creator->getId())) {
+            $creator = $this->creatorRepository->save($creator);
+        }
+        $asset->setCreatorId($creator->getId());
+
+        $this->assetRepository->save($asset);
```

Comment:
Please set `is_licensed` asset attribute to `true`

Reply:
Thank you for the suggestion. 
Since the `SaveAsset` service is common for both preview and image, I've set the flag in the corresponding controller once the image is licensed

https://github.com/magento/adobe-stock-integration/pull/411/commits/18ee4077973b4b0c7e41977ca53e9abe0d21f978